apps/test/solutions/applab/ec_turtle.js

```diff
@@ -21,7 +21,7 @@ module.exports = {
         // room to add tests here
         assert($('#screen1'));
         assert($('#turtleImage'));
-        assert.equal($('#divApplab .screen').length, 1);
+        assert.equal($('#divApplab > .screen').length, 1);
```

Comment:
Sounds good to me.  Just wanted to make sure it was considered. :+1:

On Fri, Feb 19, 2016 at 1:00 PM, David Bailey <notifications@github.com>
wrote:

> In apps/test/solutions/applab/ec_turtle.js
> <https://github.com/code-dot-org/code-dot-org/pull/6878#discussion_r53517465>
> :
>
> > @@ -21,7 +21,7 @@ module.exports = {
> >          // room to add tests here
> >          assert($('#screen1'));
> >          assert($('#turtleImage'));
> > -        assert.equal($('#divApplab .screen').length, 1);
> > +        assert.equal($('#divApplab > .screen').length, 1);
>
> Since Applab.getScreens is part of the codepath that's being tested, I'd
> rather not also use it in the test code itself because (1) it seems like a
> pattern that could conceal a bug in the utility method, and (2) it makes
> the test more fragile since if that class (part of the implementation)
> changes without changing actual behavior then it will break the test.
>
> I couldn't find this listed as an anti-pattern anywhere so let me know if
> this doesn't sound right.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/6878/files#r53517465>.
>



-- 
Brad Buchanan
Software Engineer, Code.org
brad@code.org